main.tf

```diff
@@ -172,9 +208,19 @@ resource "aws_config_configuration_aggregator" "this" {
   count = local.enabled && local.is_central_account && local.is_global_recorder_region ? 1 : 0
 
   name = module.aws_config_aggregator_label.id
-  account_aggregation_source {
-    account_ids = local.child_resource_collector_accounts
-    all_regions = true
+  dynamic "account_aggregation_source" {
+    for_each = var.aggregate_organization_wide ? [] : [1]
+    content {
+      account_ids = local.child_resource_collector_accounts
+      all_regions = true
+    }
+  }
+  dynamic "organization_aggregation_source" {
+    for_each = var.aggregate_organization_wide ? [1] : []
+    content {
+      all_regions = true
+      role_arn    = local.create_iam_role ? module.iam_role_organization_aggregator[0].arn : var.iam_role_organization_aggregator_arn
+    }
   }
```

Comment:
As this is a generic module, I'd look for a way to disable aggregation all together if the consumer doesn't want to aggregate. The way your change is currently structured, it requires aggregation, which way not be what is desired.